README.md

```diff
@@ -0,0 +1,163 @@
+* Waits for withdrawal attestations to be available and obtains them via Oracle API
+* Calls `WormholeOracleAuth.requestMint(WormholeGUID wormholeGUID, bytes signatures, uint256 maxFee)` which will:
+  * Check if `sender` is `operator` 
+  *   Check if enough valid attestations (sigs) are provided
+  *   Call `WormholeJoin.requestMint(wormholeGUID, maxfee)` which will
+        * Check if this wormhole hasn't been used before
+        * Check if the debt ceiling hasn't been reached
+        * Check the current fee via `WormholeFees`
+        * `vat.slip`, `vat.frob`, `daiJoin.exit`
+
+### Settlement
+
+Settlement process moves DAI from L1 Bridge to WormholeJoin to clear the debt that accumulates there. It is triggered by keepers.
+
+* On L2 keeper calls `l2bridge.flush()`
+* L2 -> L1 message `finalizeFlush()` is sent to `L1Bridge` and relayed by a keeper
+* `L1Bridge` upon receiving `finalizeFlush()` calls `WormholeRouter.settle()` which will
+    * Transfer `DAI` from bridges' escrow to `WormholeJoin`
+    * Call `WormholeJoin.settle()` which will use transfered DAI to clear any outstanding debt by calling `daiJoin.join`, `vat.frob`, `vat.slip`
+
+### Slow (emergency) path
+
+If attestations cannot be obtained (Oracles down or censoring), user needs to wait so that L2 message is confirmed on L1 (on Optimistic Rollups that typically is 7 days, on zkRollups it can be anything between few hours to a day). Once L2->L1 message can be relayed, user:
+
+* Relays `finalizeRegisterWormhole()`  message to `L1Bridge`
+* `L1Bridge` upon receiving `finalizeRegisterWormhole()` will call `requestMint()` on `WormholeRouter` which will:
+    * Call `WormholeJoin.requestMint(wormholeGUID, maxfee)` which will
+        * Check if this wormhole hasn't been used before
+        * Check if the debt ceiling hasn't been reached
+        * Check the current fee via `WormholeFees`
+        * `vat.slip`, `vat.frob`, `daiJoin.exit`
+
+## DAI Wormhole L2→L2
+
+![Wormhole](./docs/l2.png?raw=true)
+
+### Normal (fast) path
+
+Wormholing DAI to another L2 domain is very similar, the only difference is that DAI is minted on a target Domain rather then on L1. For this scheme to work MakerDAO `MCD` sytem needs to be deployed on a target domain. 
+
+### Settlement
+
+Settlement process is very similar, however DAI is transfered from source domain bridge on L1 to target domain bridge on L1 before rather then moved to `L1 MCD` to pay the debt. This DAI, now in target domain bridge will be backing DAI that is minted on L2 target domain.
+
+### Slow (emergency) path
+
+For a slow path, once L2->L1 message from the source domain is received on L1 and can be relayed, user can relay the message which fill call `finalizeRequestMing()` on the target domain `L1Bridge`. This will pass L1->L2 message to `L2bridge` which will call `requestMint()` on a `WormholeJoin` contract on target domain L2.
```

Comment:
```suggestion
For a slow path, once the L2->L1 message from the source domain is received on L1 and can be relayed, the user can relay the message, which will call `requestMint()` on the target domain `L1Bridge`. This will pass an L1->L2 message to `L2bridge` which will call `requestMint()` on a `WormholeJoin` contract on target domain L2.
```